File: reactjs.ts/app/models/spending.ts
import { IModel } from "./imodel";

export class Spending implements IModel {

    constructor() {
        this.Id = 0;
        this.Date = "";
        this.TagId = 0;
        this.Description = "";
        this.Price = 0;
    }

    public Id: number;
    public Description: string;
    public Date: string;
    public Price: number;
    public TagId: number;
}

File: Models/Spending.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace SpendingsCalculator.Models
{
    public class Spending : IModel
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public string Description { get; set; }
        public DateTime Date { get; set; }
        public DateTime CreateDate { get; set; }
        public float Price { get; set; }

        [ForeignKey("Tag")]
        public int TagId { get; set; }
        public Tag Tag { get; set; }
    }
}